File: practica2_845067_840710/c++/main.cc
/**
 * Autor: Jorge Jaime Modrego   Nip: 845067 
 * Autor: Javier Julve Yubero       Nip: 840710
 *            
 * Coms: En este fichero se encuentra el "Main". En el
 * se crean y se mandan a ejecutar los 3 programas test
**/
#include <stack>
#include <iostream>
#include "sumaProg.h"
#include "cuentaAtras.h"
#include "factorial.h"

// Programa principal en el que se listan y se ejecutan las instrucciones 
// de los 3 programas de la práctica
int main(){
    std::stack<int>* pila = new std::stack<int>;
    std::cout << "Ejecución de la suma de dos números" << std::endl;
    std::cout << "-----------------------------------" << std::endl;
    Suma sum(pila);
    sum.listarse(std::cout);
    std::cout << std::endl;
    sum.ejecutar(std::cout);
    
    std::cout << std::endl << std::endl << "Ejecución de la cuenta atrás" << std::endl;
    std::cout << "-----------------------------------" << std::endl;
    Cuenta cuenta(pila);
    cuenta.listarse(std::cout);
    std::cout << std::endl;
    cuenta.ejecutar(std::cout);



    std::cout << std::endl << std::endl << "Ejecución del factorial" << std::endl;
    std::cout << "-----------------------------------" << std::endl;
    Factorial fact(pila);
    fact.listarse(std::cout);
    std::cout << std::endl;
    fact.ejecutar(std::cout);
}

File: practica2_845067_840710/java/Main.java
/**
 * Autor: Alejandro Benedi Andrés   Nip: 843826 
 * Autor: Javier Julve Yubero       Nip: 840710
 *            
 * Coms: En este fichero se encuentra el "Main". En el
 * se crean y se mandan a ejecutar los 3 programas test
**/

import java.util.Scanner;
import Programas.*;

public class Main {
    public static void main(String[] args){

        Scanner sc_main = new Scanner(System.in);         // Abrimos el scanner que se va usar en los 3 programas

        Sumar p1 = new Sumar(4, sc_main);               // Crea el primer programa Suma
        p1.ejecutar_programa();                           // Ejecuta Suma
        System.out.println();
        Cuenta_atras p2 = new Cuenta_atras(7, sc_main); // Crea el segundo programa Cuenta atras
        p2.ejecutar_programa();                           // Ejecuta Cuenta atras
        System.out.println();
        Factorial p3 = new Factorial(14, sc_main);     // Crea el tercer programa Factorial
        p3.ejecutar_programa();                          // Ejecuta Factorial

        sc_main.close();                                 // Cerramos scanner
    }
}
